### ISSUE TITLE
(Feature Request) Export to JSON

### ISSUE THREAD

**Issue by rik-bos** [ArjenLammers/oqlmodule#9]
Hi Arjen,

I added a Java action that will do an export to JSON (based on how you handled the export to CSV). I also updated the Test microflow.

Would you be able to add that to the module in a new app store release?

Kind regards,
Rik Bos
[pullrequest.zip](https://github.com/ArjenLammers/oqlmodule/files/3923345/pullrequest.zip)

**Comment by ArjenLammers** [comment]
Hi Rik,

I'm having a hard time determining the added value compared to executing a OQL query and mapping it to a String with the default Export JSON functionality of Mendix.
Especially because:
 - It's not a streaming implementation (so prone to OOM's).
 - Includes custom time formatting (which is usually handled within a mapping / microflow source).

Could you explain what the added value is?

Kind regards,
Arjen Lammers

**Comment by rik-bos** [comment]
Hi Arjen,

I use it for some custom widget implementations. For me the user experience on 15k objects is better without the Mendix objects in the middle and it saves the work of creating a MD and an Export mapping. I can see that the added value would not always be very large. 

Regarding the custom date formatting, that should not be the case. The dates are formatted as normal ISO date strings, so I think that that shouldn't be a large issue.

Kind regards,
Rik

**Comment by TimvanSteenbergen** [comment]
Rik, many thanks! I will use this for years to come to generate a JSON whenever I need it, mainly getting sample data. The use case today is a masterdata page having an entity that has several obligatory objects. So I manually maintain them once, run this ExportOQLToJSON to get the JSON, create the import-mapping and run that on startup. Very helpful. Question to Arjen: What is the default Export JSON functionality in Mendix? Where can I find that?

**Comment by rik-bos** [comment]
Hi Tim, you should be able to find that here: https://docs.mendix.com/refguide/export-mapping-action/

**Comment by TimvanSteenbergen** [comment]
> 

Yeah, that's the design-time version. Like I expected, there is no runtime version, like this Export-to-JSON, that I need on regular basis, where I use an OQL for retrieving existing data to generate a JSON that I can use in my example-data. So again: greatmany thanks for your addition.
